From 9070bc210b2ecff641b73e4ade30040c1461969c Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 3 May 2023 18:31:57 +0200
Subject: [PATCH] configure.ac: fix C++11 support check

* with -Werror=return-type in CFLAGS this test fails with:

  configure:4290: checking for C++11 support or better
  configure:4303: x86_64-webos-linux-g++  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -Werror=return-type --sysroot=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/recipe-sysroot -c  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/rdfind-1.5.0=/usr/src/debug/rdfind/1.5.0-r0  -fdebug-prefix-map=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/rdfind-1.5.0=/usr/src/debug/rdfind/1.5.0-r0  -fmacro-prefix-map=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/build=/usr/src/debug/rdfind/1.5.0-r0  -fdebug-prefix-map=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/build=/usr/src/debug/rdfind/1.5.0-r0  -fdebug-prefix-map=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/recipe-sysroot=  -fmacro-prefix-map=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/recipe-sysroot=  -fdebug-prefix-map=/OE/lge/build/webos/mickledore/BUILD/work/qemux86_64-webos-linux/rdfind/1.5.0-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden   conftest.cpp >&5
  conftest.cpp: In function 'int f()':
  conftest.cpp:22:20: error: no return statement in function returning non-void [-Werror=return-type]
     22 | int f() { auto a=1;}                                                                                                                                                                                                                        |                    ^
  cc1plus: some warnings being treated as errors
  ...
  configure:4308: error: no c++11 support, please set CXXFLAGS properly

* fix the test to pass

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Upstream-Status: Submitted [https://github.com/pauldreik/rdfind/pull/132]
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index be1b2fd..9c3513c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -64,7 +64,7 @@ AC_SYS_LARGEFILE
 
 dnl make sure we have c++11 or better,
 AC_MSG_CHECKING([for C++11 support or better])
-AC_COMPILE_IFELSE([AC_LANG_PROGRAM([int f() { auto a=1;}])],
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([int f() { auto a=1;return a;}])],
                   [AC_MSG_RESULT([yes])],
 		  [AC_MSG_ERROR([no c++11 support, please set CXXFLAGS properly])])
 
